### PR TITLE
SPaT Code Coverage

### PR DESCRIPTION
**Pull request by drewjj**
This adds complete code coverage for SPaT.

Sample data was created through erlang and converted to XML and JSON with the asn1c scripts. Some tests directly target corner cases where natural ser/des does not cover.

image

### CODE REVIEW

File: j2735-2024/src/test/java/j2735/SPAT/IntersectionStatusObjectTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package j2735.SPAT;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit tests for the IntersectionStatusObject class.
+ * Generated using ChatGPT
+ */
+public class IntersectionStatusObjectTest {
```
Comment (iyourshaw) [inline j2735-2024/src/test/java/j2735/SPAT/IntersectionStatusObjectTest.java:10]: For bitstrings prefer testing the actual function they perform, which is to deserialize from text, rather than setting each bit programmatically and testing that it was set, which is very verbose for humans to try to review.  See the MapData LaneAttributes test for some examples.

**Review by iyourshaw — CHANGES_REQUESTED** [review]
Looks good, but please redo the bitstring test as noted.

File: j2735-2024/src/test/java/j2735/SPAT/IntersectionStatusObjectTest.java (hunk 1 above)
Comment (drewjj) [reply j2735-2024/src/test/java/j2735/SPAT/IntersectionStatusObjectTest.java:10]: Addressed

**Review by iyourshaw — APPROVED** [review]
all looks good
